test-monitor/nitestmonitor.yml

```diff
@@ -1150,33 +1170,43 @@ paths:
           description: Not Found
           schema:
             $ref: '#/definitions/Error'
-  /v1/query-results:
+  /v2/query-results:
```

Comment:
Our current guidelines mean you also need to make v2 of all of the other routes as well. Either by copying them all (eww) or making the version a parameter like for the /{version} route.

Reply:
Copying all seems a bit extreme. An improvement we can make to our API versioning is to set a parameter in the header of the request, thus leaving the route as-is and clients would explicitly request a specific version of the route. 
Thoughts?

Reply:
We're currently investigating this process:

- For routes that break compatibility, copy the current version of the route and give it the deprecated tag, then make a vnext version of the route with the change.
- All other routes that are still compatible, bump to vnext and add an attribute indicating it is still compatible with a previous range of versions. For example, something like `x-ni-oldest-compatible-version: 1`.
- Bump the service version at the top in the info section to vnext.

Then we can use the compatibility attribute during codegen to automatically duplicate the routes in the INI file without needing to duplicate them in the documentation or the code.

Reply:
I would like to avoid having to duplicate generated code (or docs) for routes which maintained backwards compatibility across the version change. There's an item on our backlog to make the necessary codegen changes. I'll keep this review alive until that is complete.

Reply:
The codegen changes are up for review now.